0000-payment-pointers/0000-payment-pointers.md

```diff
@@ -0,0 +1,126 @@
+
+### Flexible
+
+The payment pointer should not be tightly couple to a specific payment setup protocol or protocol version. It should be possible for any new payment setup protocol to leverage payment pointers.
+
+### Widely Usable
+
+It should be simple for individuals or small companies to host their own payment setup protocol receiver endpoints at a URL that can be resolved via a simple and easily recognizable payment pointer. Likewise, it should be possible for a payment services provider to host payment setup protocol receiver endpoints for multiple entities without the risk of hosting them at endpoint URLs that conflict with their other services. To that end the provider should have the option of hosting different receivier endpoints under the same domain and a different path or at a different subdomain for each receiver.
+
+## Payment Setup Protocols
+
+A payment setup protocol is defined very generically as a protocol for exchanging payment information between a sender and receiver before executing the payment.
+
+An example of a payment setup protocol is the [Simple Payment Setup Protocol](https://interledger.org/rfcs/0009-simple-payment-setup-protocol/) used to setup an Interledger payment.
+
+Payment pointers MUST resolve to an endpoint hosted by a payment receiver at which a payment setup protocol can be initiated. This endpoint is known as the payment setup protocol receiver endpoint.
+
+A payment setup protocol receiver endpoint is identified by an https URL as defined in [RFC7230](https://tools.ietf.org/html/rfc7230#section-2.7.2). It MUST accept HTTP GET requests to initiate the protocol.
+
+Payment setup protocols SHOULD define a custom MIME type for their message definitions and use HTTP content negotitation between the client and endpoint server to determine the payment setup protocol and protocol versions supported by the requesting client and the endpoint server.
+
+## Payment Pointer Syntax
+
+This specification uses the Augmented Backus-Naur Form (ABNF) notation of [RFC2234](https://tools.ietf.org/html/rfc2234) and imports the following rules from [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.3)): _host_ and _path-abempty_.
+
+The syntax of a payment pointer is:
+
+```
+  "$" host path-abempty
+```
+
+## Payment Setup Protocol Receiver Endpoint Resolution
+
+Given a payment pointer of the form `"$" host path-abempty` the payment setup protocol receiver endpoint URL is:
+
+```
+  "https://" host path-abempty
+```
+
+If _path-abempty_ is empty then it is assigned the value `/.well-known/pay`.
+
+Note that this is a restrcited profile of the data allowed in a full https URL. Where the URL syntax supports an _authority_ the payment pointer syntax only supports a _host_ which excludes the _userinfo_ and _port_. The payment pointer syntax also excludes the _query_ and _fragment_ parts that are allowed in the URL syntax.
+
```

Comment:
Mention that you can have a redirect, but maybe clients shouldn't follow more than 1 (or 2?) redirects